hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AFileSystem.java

```diff
@@ -4532,8 +4814,10 @@ private HeaderProcessing getHeaderProcessing() {
   public RemoteIterator<LocatedFileStatus> listLocatedStatus(final Path f,
       final PathFilter filter)
       throws FileNotFoundException, IOException {
-    entryPoint(INVOCATION_LIST_LOCATED_STATUS);
     Path path = qualify(f);
+    // Unless that iterator is closed, the iterator wouldn't be closed
+    // there.
+    entryPoint(INVOCATION_LIST_LOCATED_STATUS, path);
```

Comment:
you are not using try with resource here. maybe it's justified, I'm just pointing it out because maybe it's needed.

Reply:
I try to explain it above. The iterator is picking it up. We do the same for all ops which return something (listing, input stream, output stream...) which outlives the API. The audit span goes with the object and so is left open. Listings are a troublespot as they are unlikely ever to be closed.

Reply:
Latest PR is moving back to try-with-resource/some closure mechanism. As close() only deactivates the span, it does not close it.